src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/CSharpCodeParser.cs

```diff
@@ -1459,6 +1470,52 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                                     return;
                                 }
 
+                                break;
+                            case DirectiveTokenKind.GenericTypeConstraint:
+                                if (At(SyntaxKind.Keyword) &&
+                                    string.Equals(CurrentToken.Content, CSharpLanguageCharacteristics.GetKeyword(CSharpKeyword.Where), StringComparison.Ordinal))
+                                {
+                                    // Consume the 'where' keyword plus any aditional whitespace
+                                    AcceptAndMoveNext();
+                                    AcceptWhile(SyntaxKind.Whitespace);
+                                    // Check that the type name matches the type name before the where clause.
+                                    // Find a better way to do this
```

Comment:
What happens if a user does something like `@typeparam where` without providing the parameter name?

Reply:
That is mapped to a generic type parameter with name where and the razor compiler produces an error (which is expected).